NPE when use 'message.labelIds'

```
[SUMMARY]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
java.lang.NullPointerException: historyLabelRemoved.message.labelIds must not be null


[STACK_TRACE]
com.flowcrypt.email.util.exception.ManualHandledException: Handled manually:
        at com.flowcrypt.email.util.exception.ExceptionUtil$Companion.handleError(ExceptionUtil.kt:27)
        at com.flowcrypt.email.api.email.gmail.GmailApiHelper$Companion$executeWithResult$2.invokeSuspend(GmailApiHelper.kt:142)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.UndispatchedCoroutine.afterResume(CoroutineContext.kt:233)
        at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:102)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
        at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.lang.NullPointerException: historyLabelRemoved.message.labelIds must not be null
        at com.flowcrypt.email.api.email.gmail.GmailApiHelper$Companion$processHistory$2.invokeSuspend(GmailApiHelper.kt:572)
        at com.flowcrypt.email.api.email.gmail.GmailApiHelper$Companion$processHistory$2.invoke(Unknown Source:8)
        at com.flowcrypt.email.api.email.gmail.GmailApiHelper$Companion$processHistory$2.invoke(Unknown Source:4)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at com.flowcrypt.email.api.email.gmail.GmailApiHelper$Companion.processHistory(GmailApiHelper.kt:525)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker$handleMsgsFromHistory$2.invokeSuspend(InboxIdleSyncWorker.kt:143)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker$handleMsgsFromHistory$2.invoke(Unknown Source:8)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker$handleMsgsFromHistory$2.invoke(Unknown Source:4)
        at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
        at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
        at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker.handleMsgsFromHistory(InboxIdleSyncWorker.kt:142)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker.access$handleMsgsFromHistory(InboxIdleSyncWorker.kt:39)
        at com.flowcrypt.email.jetpack.workmanager.sync.InboxIdleSyncWorker$syncMessages$4$1.invokeSuspend(InboxIdleSyncWorker.kt:134)
        ... 11 more
```

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/api/email/gmail/GmailApiHelper.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/api/email/gmail/GmailApiHelper.kt
@@ -377,15 +377,17 @@ class GmailApiHelper {
     suspend fun deleteMsgsPermanently(
       context: Context, accountEntity: AccountEntity,
       ids: List<String>
-    ) = withContext(Dispatchers.IO) {
-      val gmailApiService = generateGmailApiService(context, accountEntity)
-      gmailApiService
-        .users()
-        .messages()
-        .batchDelete(DEFAULT_USER_ID, BatchDeleteMessagesRequest().apply {
-          this.ids = ids
-        })
-        .execute()
+    ) {
+      withContext(Dispatchers.IO) {
+        val gmailApiService = generateGmailApiService(context, accountEntity)
+        gmailApiService
+          .users()
+          .messages()
+          .batchDelete(DEFAULT_USER_ID, BatchDeleteMessagesRequest().apply {
+            this.ids = ids
+          })
+          .execute()
+      }
     }
 
     suspend fun deleteDrafts(
@@ -540,7 +542,7 @@ class GmailApiHelper {
 
         history.messagesAdded?.let { messagesAdded ->
           for (historyMsgAdded in messagesAdded) {
-            if (LABEL_DRAFT in historyMsgAdded.message.labelIds && !isDrafts) {
+            if (LABEL_DRAFT in (historyMsgAdded.message.labelIds ?: emptyList()) && !isDrafts) {
               //skip adding drafts to non-Drafts folder
               continue
             }
@@ -552,16 +554,16 @@ class GmailApiHelper {
 
         history.labelsRemoved?.let { labelsRemoved ->
           for (historyLabelRemoved in labelsRemoved) {
-            if (localFolder.fullName in historyLabelRemoved.labelIds) {
+            if (localFolder.fullName in (historyLabelRemoved.labelIds ?: emptyList())) {
               newCandidatesMap.remove(historyLabelRemoved.message.uid)
               updateCandidates.remove(historyLabelRemoved.message.uid)
               deleteCandidatesUIDs.add(historyLabelRemoved.message.uid)
               continue
             }
 
-            if (LABEL_TRASH in historyLabelRemoved.labelIds) {
+            if (LABEL_TRASH in (historyLabelRemoved.labelIds ?: emptyList())) {
               val msg = historyLabelRemoved.message
-              if (localFolder.fullName in msg.labelIds) {
+              if (localFolder.fullName in (msg.labelIds ?: emptyList())) {
                 deleteCandidatesUIDs.remove(msg.uid)
                 updateCandidates.remove(msg.uid)
                 newCandidatesMap[msg.uid] = msg
@@ -569,28 +571,29 @@ class GmailApiHelper {
               }
             }
 
-            val existedFlags = labelsToImapFlags(historyLabelRemoved.message.labelIds)
+            val existedFlags =
+              labelsToImapFlags(historyLabelRemoved.message.labelIds ?: emptyList())
             updateCandidates[historyLabelRemoved.message.uid] = existedFlags
           }
         }
 
         history.labelsAdded?.let { labelsAdded ->
           for (historyLabelAdded in labelsAdded) {
-            if (localFolder.fullName in historyLabelAdded.labelIds) {
+            if (localFolder.fullName in (historyLabelAdded.labelIds ?: emptyList())) {
               deleteCandidatesUIDs.remove(historyLabelAdded.message.uid)
               updateCandidates.remove(historyLabelAdded.message.uid)
               newCandidatesMap[historyLabelAdded.message.uid] = historyLabelAdded.message
               continue
             }
 
-            if (historyLabelAdded.labelIds.contains(LABEL_TRASH)) {
+            if ((historyLabelAdded.labelIds ?: emptyList()).contains(LABEL_TRASH)) {
               newCandidatesMap.remove(historyLabelAdded.message.uid)
               updateCandidates.remove(historyLabelAdded.message.uid)
               deleteCandidatesUIDs.add(historyLabelAdded.message.uid)
               continue
             }
 
-            val existedFlags = labelsToImapFlags(historyLabelAdded.message.labelIds)
+            val existedFlags = labelsToImapFlags(historyLabelAdded.message.labelIds ?: emptyList())
             updateCandidates[historyLabelAdded.message.uid] = existedFlags
           }
         }
@@ -1019,7 +1022,7 @@ class GmailApiHelper {
         request.fields = fields.joinToString(separator = ",")
       }
 
-      return@withContext request.execute().drafts
+      return@withContext request.execute().drafts ?: emptyList()
     }
 
     /**
@@ -1037,7 +1040,7 @@ class GmailApiHelper {
         .setSelectedAccount(account)
     }
 
-    private fun labelsToImapFlags(labelIds: MutableList<String>): Flags {
+    private fun labelsToImapFlags(labelIds: List<String>): Flags {
       val flags = Flags()
       labelIds.forEach {
         when (it) {

--- a/FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/MessagesViewModel.kt
+++ b/FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/MessagesViewModel.kt
@@ -535,7 +535,7 @@ class MessagesViewModel(application: Application) : AccountViewModel(application
     GmailApiHelper.identifyAttachments(msgEntities, msgs, account, localFolder, roomDatabase)
     val session = Session.getInstance(Properties())
     updateLocalContactsIfNeeded(messages = msgs
-      .filter { it.labelIds.contains(GmailApiHelper.LABEL_SENT) }
+      .filter { it.labelIds?.contains(GmailApiHelper.LABEL_SENT) == true }
       .map { GmaiAPIMimeMessage(session, it) }.toTypedArray()
     )
   }
@@ -809,7 +809,7 @@ class MessagesViewModel(application: Application) : AccountViewModel(application
     GmailApiHelper.identifyAttachments(msgEntities, msgs, account, localFolder, roomDatabase)
     val session = Session.getInstance(Properties())
     updateLocalContactsIfNeeded(messages = msgs
-      .filter { it.labelIds.contains(GmailApiHelper.LABEL_SENT) }
+      .filter { it.labelIds?.contains(GmailApiHelper.LABEL_SENT) == true }
       .map { GmaiAPIMimeMessage(session, it) }.toTypedArray()
     )
   }
@@ -1035,7 +1035,7 @@ class MessagesViewModel(application: Application) : AccountViewModel(application
       if (folderType === FoldersManager.FolderType.SENT) {
         val session = Session.getInstance(Properties())
         updateLocalContactsIfNeeded(messages = newCandidates
-          .filter { it.labelIds.contains(GmailApiHelper.LABEL_SENT) }
+          .filter { it.labelIds?.contains(GmailApiHelper.LABEL_SENT) == true }
           .map { GmaiAPIMimeMessage(session, it) }.toTypedArray()
         )
       }

